.github/workflows/uberjar.yml

```diff
@@ -75,44 +158,49 @@ jobs:
           echo "${{ steps.info.outputs.result }}"
 
   build:
-    needs: get-container-info
+    needs:
+      [check-existing-artifacts, get-build-requirements, get-container-info]
+    if: always() && !cancelled() && needs.check-existing-artifacts.outputs.should_build == 'true' && (needs.get-build-requirements.result == 'success' || needs.get-build-requirements.result == 'skipped')
     name: Build MB ${{ matrix.edition }}
     runs-on: ubuntu-22.04
     timeout-minutes: 15
     strategy:
       matrix:
-        edition: ${{ fromJson(needs.get-container-info.outputs.editions) }}
+        edition: ${{ needs.get-container-info.result == 'success' && fromJson(needs.get-container-info.outputs.editions) || fromJson('["ee", "oss"]') }}
     env:
       MB_EDITION: ${{ matrix.edition }}
       INTERACTIVE: false
-      SKIP_LICENSES: true # faster builds for dev
+      SKIP_LICENSES: ${{ github.event_name == 'pull_request' }}
     steps:
       - name: Check out the code
         uses: actions/checkout@v4
         with:
           ref: ${{ github.event.inputs.commit }}
       - name: Prepare front-end environment
         uses: ./.github/actions/prepare-frontend
+        with:
+          node-version: "${{ needs.get-build-requirements.outputs.node_version }}"
```

Comment:
should there be a fallback version here?

Reply:
Nice catch! Done!